### ISSUE TITLE
undefined reference to `_Unwind_Resume' when linking with clang

### ISSUE THREAD

**Issue by newincpp** [armory3d/Kromx#9]
Hi,
I cloned and followed the steps on armoryPaint README.md but unfortunatly I get errors with both clang and gcc.
I do have an up-to-date libcxx (10.0.0) and an updated clang (also 10.0.0)
But when the linking of Krom happen I get a few hundreds line like:
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_alloc.o): in function `(anonymous namespace)::pool::free(void*) [clone .constprop.0] [clone .cold]':
(.text.unlikely._ZN12_GLOBAL__N_14pool4freeEPv.constprop.0+0x4a): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_alloc.o): in function `(anonymous namespace)::pool::allocate(unsigned long) [clone .constprop.0] [clone .cold]':
(.text.unlikely._ZN12_GLOBAL__N_14pool8allocateEm.constprop.0+0x4a): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_personality.o): in function `base_of_encoded_value(unsigned char, _Unwind_Context*)':
(.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x2d): undefined reference to `_Unwind_GetTextRelBase'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x4e): undefined reference to `_Unwind_GetRegionStart'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text._ZL21base_of_encoded_valuehP15_Unwind_Context+0x98): undefined reference to `_Unwind_GetDataRelBase'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_personality.o): in function `parse_lsda_header(_Unwind_Context*, unsigned char const*, lsda_header_info*)':
(.text._ZL17parse_lsda_headerP15_Unwind_ContextPKhP16lsda_header_info+0x1b): undefined reference to `_Unwind_GetRegionStart'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_personality.o): in function `__gxx_personality_v0':
(.text.__gxx_personality_v0+0xba): undefined reference to `_Unwind_SetGR'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.__gxx_personality_v0+0xca): undefined reference to `_Unwind_SetGR'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.__gxx_personality_v0+0xd5): undefined reference to `_Unwind_SetIP'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.__gxx_personality_v0+0x114): undefined reference to `_Unwind_GetLanguageSpecificData'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.__gxx_personality_v0+0x164): undefined reference to `_Unwind_GetIPInfo'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(eh_personality.o): in function `__cxa_call_unexpected.cold':
(.text.unlikely.__cxa_call_unexpected+0xa1): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(vterminate.o): in function `__gnu_cxx::__verbose_terminate_handler() [clone .cold]':
(.text.unlikely._ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x86): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/libstdc++.a(ios_init.o): in function `std::ios_base::Init::Init() [clone .cold]':
(.text.unlikely._ZNSt8ios_base4InitC2Ev+0x1f): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.unlikely._ZNSt8ios_base4InitC2Ev+0x42): undefined reference to `_Unwind_Resume'
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: (.text.unlikely._ZNSt8ios_base4InitC2Ev+0x65): undefined reference to `_Unwind_Resume'
```
There is plenty more but you get the idea.
Your build system looks fairly unconventional so I was wondering if you know where I should look at to guess what is mismatching between my (custom build) libcxx/clang/llvm and the one you used.

**Comment by luboslenco** [comment]
Hi, sorry but I am unable to tell what is wrong. My guess is v8 is picky about libcxx and something throws it off. If it helps Linux binaries for ArmorPaint are compiled on `Ubuntu 18.04.3 LTS` and `clang -v` reads `clang version 6.0.0-1ubuntu2`.

**Comment by newincpp** [comment]
It was partly my fault and I fixed a bunch of link errors (by recompiling my clang with libc++ enabled by default). I don't have the `undefined reference to `_Unwind_GetRegionStart'` errors.
I also removed all `static-libgcc` and `static-libstdc++` of the build system and it fixed a whole bunch of errors.
The only remaing one are linking all related to linking with v8 but I'm not ready to make cohabiting 2 clangs in my distros so I'm not sure if I should go in the territory of compiling my own v8 and replace the current one or forgetting about compiling armorpaint.
Though as the original problem is fixed I'm gonna close this issue.
